cmd/snap-seccomp: use i386 cross-compiler instead of multilib

zyga opened the pull request:
Multilib gcc has been deprecated in Debian. Instead we are using a i386 cross-compiler in the seccomp tests. The change itself is non-controversial, but we need to address the packaging where we pretend packaging/ubuntu-16.04 is really working on 16.04 systems.

ernestl commented on cmd/snap-seccomp/main_test.go:
```diff
@@ -226,8 +226,7 @@ func (s *snapSeccompSuite) SetUpSuite(c *C) {
 	// Ideally we would build for ppc64el->powerpc and arm64->armhf but
 	// it seems tricky to find the right gcc-multilib for this.
```
L227 and L236: remove reference to `gcc-multilib`

ernestl reviewed (commented):
Thanks, nitpick comment, CLA and unit test failure to look at.

zyga commented on build-aux/snap/snapcraft.yaml:
```diff
@@ -245,6 +245,9 @@ parts:
       - on arm64:
         - optee-client-dev
+      - on amd64:
+        - libc6-dev-i386-cross
+        - gcc-i686-linux-gnu
```
I didn't realize we are not building tests at all.

zyga replied:
Dropped.

bboozzoo commented on .image-garden.mk:
```diff
@@ -32,6 +32,8 @@ $(CLOUD_INIT_USER_DATA_TEMPLATE)
 - sed -i -e 's/^SELINUX=enforcing/SELINUX=disabled/g' /etc/selinux/config
 - update-bootloader
+packages:
+- apparmor-utils
```
unrelated?

ernestl reviewed (approved):
Thanks

zyga committed:
tests: remove broken dumper part

This part doesn't build as the "files" source is gone.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

zyga committed:
cmd/snap-seccomp: use i386 cross-compiler or -m32

Multi-lib gcc has been deprecated in Debian. Instead, where possible, are using
a i386 cross-compiler for seccomp tests. Debian packaging had been adjusted to
use the cross compiler in the downstream 2.71 release. Ubuntu "16.04"
packaging, which is really not for 16.04, is adjusted to match.

The test for snapd-daemon-user is WIP

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>
